File: BotsSDK/korebot/src/main/java/kore/botssdk/utils/ClosingService.java
package kore.botssdk.utils;

import android.app.Service;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Build;
import android.os.IBinder;

import androidx.annotation.Nullable;

import kore.botssdk.models.BotResponse;

public class ClosingService extends Service {

    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        return null;
    }

    @Override
    public void onTaskRemoved(Intent rootIntent) {

        // Handle application closing
        Intent intent = new Intent(BundleConstants.DESTROY_EVENT);
        sendBroadcast(intent);

        if (Build.VERSION.SDK_INT > Build.VERSION_CODES.TIRAMISU) {
            SharedPreferences.Editor editor = getSharedPreferences(BotResponse.THEME_NAME, Context.MODE_PRIVATE).edit();
            editor.putBoolean(BundleConstants.IS_RECONNECT, false);
            editor.putInt(BotResponse.HISTORY_COUNT, 0);
            editor.apply();
        }

        LogUtils.e("onTaskRemoved", "onTaskRemoved called");
        // Destroy the service
        stopSelf();
        super.onTaskRemoved(rootIntent);
    }
}

File: BotSdk/ui/src/main/java/com/kore/services/ClosingService.kt
package com.kore.services

import android.app.Service
import android.content.Intent
import android.os.Build
import android.os.IBinder
import com.kore.common.utils.LogUtils
import com.kore.constants.SharedPrefConstants.HISTORY_COUNT
import com.kore.model.constants.BotResponseConstants
import com.kore.ui.utils.BundleConstants
import androidx.core.content.edit

class ClosingService : Service() {
    override fun onBind(intent: Intent): IBinder? {
        return null
    }

    override fun onTaskRemoved(rootIntent: Intent) {
        // Handle application closing
        val intent = Intent(BundleConstants.DESTROY_EVENT)
        intent.setPackage(packageName)
        sendBroadcast(intent)

        if (Build.VERSION.SDK_INT > Build.VERSION_CODES.TIRAMISU) {
            getSharedPreferences(BotResponseConstants.THEME_NAME, MODE_PRIVATE).edit { putInt(HISTORY_COUNT, 0) }
        }

        LogUtils.i("onTaskRemoved", "onTaskRemoved called")
        // Destroy the service
        stopSelf()
        super.onTaskRemoved(rootIntent)
    }
}
